src/OpenLoco/src/GameCommands/Terraform/LowerRaiseLandMountain.cpp

```diff
@@ -0,0 +1,405 @@
+    {
+        adjustSurfaceSlope(pos, targetBaseZ, 3, SurfaceSlope::CornerUp::west, removedBuildings);
+    }
+
+    // 0x0046357C
+    static void adjustSurfaceSlopeNorth(Pos2 pos, int8_t targetBaseZ, std::set<Pos3, LessThanPos3>& removedBuildings)
+    {
+        adjustSurfaceSlope(pos, targetBaseZ, 0, SurfaceSlope::CornerUp::north, removedBuildings);
+    }
+
+    // 0x0046363F
+    static void adjustSurfaceSlopeEast(Pos2 pos, int8_t targetBaseZ, std::set<Pos3, LessThanPos3>& removedBuildings)
+    {
+        adjustSurfaceSlope(pos, targetBaseZ, 1, SurfaceSlope::CornerUp::east, removedBuildings);
+    }
+
+    static uint32_t adjustMountainCentre(const LowerRaiseLandMountainArgs& args, std::set<Pos3, LessThanPos3>& removedBuildings, const uint8_t flags)
+    {
+        // Prepare parameters for raise/lower land tool
+        uint32_t result = FAILURE;
+        if (args.adjustment == 1)
+        {
+            RaiseLandArgs raiseArgs;
+            raiseArgs.centre = args.centre;
+            raiseArgs.pointA = args.pointA;
+            raiseArgs.pointB = args.pointB;
+            raiseArgs.corner = 4;
+
+            result = raiseLand(raiseArgs, removedBuildings, flags);
+        }
+        else
+        {
+            LowerLandArgs lowerArgs;
+            lowerArgs.centre = args.centre;
+            lowerArgs.pointA = args.pointA;
+            lowerArgs.pointB = args.pointB;
+            lowerArgs.corner = 4;
+
+            result = lowerLand(lowerArgs, removedBuildings, flags);
+        }
+
+        return result;
+    }
+
+    static void smoothenSurfaceNorth(const Pos2& refPos, const Pos2& targetPos, const LowerRaiseLandMountainArgs& args, std::set<Pos3, LessThanPos3>& removedBuildings)
+    {
+        auto tile = TileManager::get(refPos);
+        auto surface = tile.surface();
+
+        auto height = TileManager::getSurfaceCornerHeight(*surface, SurfaceSlope::CornerUp::south);
+        adjustSurfaceSlopeEast(targetPos, height, removedBuildings);
+
+        *mtnToolHeightDiff -= kSmallZStep;
+
+        if (targetPos.y >= args.pointA.y)
+        {
+            *mtnToolHeightDiff += kSmallZStep;
+            if (targetPos.y > args.pointB.y)
+            {
+                *mtnToolHeightDiff += kSmallZStep;
+            }
+        }
+
+        height = TileManager::getSurfaceCornerHeight(*surface, SurfaceSlope::CornerUp::west);
+        adjustSurfaceSlopeNorth(targetPos, height, removedBuildings);
+    }
+
+    static void smoothenSurfaceEast(const Pos2& refPos, const Pos2& targetPos, const LowerRaiseLandMountainArgs& args, std::set<Pos3, LessThanPos3>& removedBuildings)
+    {
+        auto tile = TileManager::get(refPos);
+        auto surface = tile.surface();
+
+        auto height = TileManager::getSurfaceCornerHeight(*surface, SurfaceSlope::CornerUp::west);
+        adjustSurfaceSlopeSouth(targetPos, height, removedBuildings);
```

Comment:
each of these 'smoothen' functions calls the method one direction clockwise here - is that correct?

Reply:
Yes, that stood out to me as well, initially. However, I triple-checked that this is what vanilla is doing as well.